### PR TITLE
Knobgoblin keyboard initial commit

### PR DESCRIPTION
**Pull request by mrT1ddl3s**
New macropad initial commit. Keyboard and keymap

## Description
This is a keyboard and keymap addition for a new board.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x ] Keyboard (addition or update)
- [x ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/knobgoblin/config.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x0000
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    MrT1ddl3s
+#define PRODUCT         knobgoblin
+#define DESCRIPTION     numpad macropad 2x encoders and oled support
```
Comment (fauxpark) [inline keyboards/knobgoblin/config.h:27]:
```suggestion
```

File: keyboards/knobgoblin/readme.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# KNOBGOBLIN
+
+![Knob Goblin](https://i.imgur.com/oYnzqqN.jpg)
```
Comment (fauxpark) [inline keyboards/knobgoblin/readme.md:3]:
```suggestion
![Knob Goblin](https://i.imgur.com/oYnzqqNl.jpg)
```

File: keyboards/knobgoblin/readme.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+A 5x4 numpad with 2x rotary encoder and OLED support. 
+
+* Keyboard Maintainer: [MrT1ddl3s](https://github.com/mrT1ddl3s)
+* Hardware Supported: knobgoblin PCB
+* Hardware Availability: Occasional group buy.[knobgoblin.xyz](https://knobgoblin.xyz)
+
+Make example for this keyboard (after setting up your build environment):
+
+	make knobgoblin:default
```
Comment (fauxpark) [inline keyboards/knobgoblin/readme.md:13]:
```suggestion
    make knobgoblin:default
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by mrT1ddl3s** [comment]
Thank you Drashna and Fauxpark. I've commited your suggested changes. Please let me know if there are any other changes to be made before merge.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**